tests/checks/test_regulation.py

```diff
@@ -164,3 +164,215 @@ def test_validate_row_constraint_list_in_formula_issue_817():
+    report = validate(
+        "data/table-limits.csv",
+        checks=[{"code": "table-dimensions"}],
+    )
+    assert report.flatten(["limits", "code"]) == []
+
+
+def test_validate_table_dimensions_num_fields_num_rows_wrong():
+    report = validate(
+        "data/table-limits.csv",
+        checks=[checks.regulation.table_dimensions(num_fields=3, num_rows=2)],
+    )
+    assert report.flatten(["limits", "code"]) == [
+        [{"required_num_fields": 3, "number_fields": 4}, "table-dimensions-error"],
+        [{"required_num_rows": 2, "number_rows": 3}, "table-dimensions-error"],
+    ]
+
+
+def test_validate_table_dimensions_num_fields_num_rows_wrong_declarative():
+    report = validate(
+        "data/table-limits.csv",
+        checks=[{"code": "table-dimensions", "num_fields": 3, "num_rows": 2}],
+    )
+    assert report.flatten(["limits", "code"]) == [
+        [{"required_num_fields": 3, "number_fields": 4}, "table-dimensions-error"],
+        [{"required_num_rows": 2, "number_rows": 3}, "table-dimensions-error"],
+    ]
+
+
+def test_validate_table_dimensions_num_fields_num_rows_correct():
+    report = validate(
+        "data/table-limits.csv",
+        checks=[checks.regulation.table_dimensions(num_fields=4, num_rows=3)],
+    )
+    assert report.flatten(["limits", "code"]) == []
+
+
+def test_validate_table_dimensions_num_fields_num_rows_correct_declarative():
+    report = validate(
+        "data/table-limits.csv",
+        checks=[{"code": "table-dimensions", "num_fields": 4, "num_rows": 3}],
+    )
+    assert report.flatten(["limits", "code"]) == []
+
+
+def test_validate_table_dimensions_min_fields_max_rows_wrong():
+    report = validate(
+        "data/table-limits.csv",
+        checks=[checks.regulation.table_dimensions(min_fields=5, max_rows=2)],
+    )
+    assert report.flatten(["limits", "code"]) == [
+        [{"min_fields": 5, "number_fields": 4}, "table-dimensions-error"],
+        [{"max_rows": 2, "number_rows": 3}, "table-dimensions-error"],
+    ]
+
+
+def test_validate_table_dimensions_min_fields_max_rows_wrong_declarative():
+    report = validate(
+        "data/table-limits.csv",
+        checks=[{"code": "table-dimensions", "min_fields": 5, "max_rows": 2}],
+    )
+    assert report.flatten(["limits", "code"]) == [
+        [{"min_fields": 5, "number_fields": 4}, "table-dimensions-error"],
+        [{"max_rows": 2, "number_rows": 3}, "table-dimensions-error"],
+    ]
+
+
+def test_validate_table_dimensions_min_fields_max_rows_correct():
+    report = validate(
+        "data/table-limits.csv",
+        checks=[checks.regulation.table_dimensions(min_fields=4, max_rows=3)],
+    )
+    assert report.flatten(["limits", "code"]) == []
+
+
+def test_validate_table_dimensions_min_fields_max_rows_correct_declarative():
+    report = validate(
+        "data/table-limits.csv",
+        checks=[{"code": "table-dimensions", "min_fields": 4, "max_rows": 3}],
```

Comment:
@aivuk 
We need to use camelCase for property naming inside declarative descriptors (I think I mentioned it somewhere previously):
- https://framework.frictionlessdata.io/docs/guides/quick-start#usage
- https://github.com/frictionlessdata/frictionless-py/blob/main/frictionless/checks/regulation.py#L52-L59

Reply:
I saw it but it was not clear if it was mandatory. I fixed now.